core/src/main/java/org/apache/iceberg/UpdateSnapshotReferencesOperation.java

```diff
@@ -0,0 +1,224 @@
+    ValidationException.check(ref.isBranch(), "Ref with name %s is a tag not a branch", name);
+    SnapshotRef updateBranch = SnapshotRef.builderFrom(ref)
+        .minSnapshotsToKeep(minSnapshotsToKeep)
+        .build();
+    refsToUpdate.put(name, updateBranch);
+    return this;
+  }
+
+  public UpdateSnapshotReferencesOperation setMaxSnapshotAgeMs(String name, long maxSnapshotAgeMs) {
+    Preconditions.checkNotNull(name, "Branch name cannot be null");
+    SnapshotRef ref = validateAndGetLatestRef(name);
+    ValidationException.check(ref != null, "No such ref with name %s", name);
+    ValidationException.check(ref.isBranch(), "Ref with name %s is a tag not a branch", name);
+    SnapshotRef updateBranch = SnapshotRef.builderFrom(ref)
+        .maxSnapshotAgeMs(maxSnapshotAgeMs)
+        .build();
+    refsToUpdate.put(name, updateBranch);
+    return this;
+  }
+
+  public UpdateSnapshotReferencesOperation setMaxRefAgeMs(String name, long maxRefAgeMs) {
+    Preconditions.checkNotNull(name, "Reference name cannot be null");
+    SnapshotRef ref = validateAndGetLatestRef(name);
+    ValidationException.check(ref != null, "No such ref with name %s", name);
+    ValidationException.check(ref.isBranch(), "Ref with name %s is a tag not a branch", name);
+    SnapshotRef updatedTag = SnapshotRef.builderFrom(ref)
+        .maxRefAgeMs(maxRefAgeMs)
+        .build();
+    refsToUpdate.put(name, updatedTag);
+    return this;
+  }
+
+  /**
+   * Helper which validates if a reference exists either in existing
+   * metadata or in the ongoing operation. If the reference is updated in the ongoing operation
+   * or if the reference does not exist in the existing metadata return that view of the reference.
+   */
+  private SnapshotRef validateAndGetLatestRef(String name) {
+    SnapshotRef ref = base.refs().get(name);
+    if (ref == null || refsToUpdate.containsKey(name)) {
+      ref = refsToUpdate.get(name);
+    }
+    ValidationException.check(ref != null, "No such ref with name %s", name);
+    ValidationException.check(ref.isBranch(), "Ref with name %s is a tag not a branch", name);
+    return ref;
+  }
+
+  private TableMetadata internalApply() {
+    this.base = ops.refresh();
+    TableMetadata.Builder updatedBuilder = TableMetadata.buildFrom(base);
+    for (String refName : refsToRemove) {
+      SnapshotRef ref = base.ref(refName);
+      ValidationException.check(ref != null, "Cannot remove nonexistent snapshot ref %s", refName);
+      if (ref.isBranch()) {
+        updatedBuilder.removeBranch(refName);
+      } else {
+        updatedBuilder.removeTag(refName);
+      }
+    }
+    Map<String, SnapshotRef> refs = Maps.newHashMap();
+    for (Map.Entry<String, SnapshotRef> refEntry : refsToUpdate.entrySet()) {
+      String name = refEntry.getKey();
+      SnapshotRef snapshotRef = refEntry.getValue();
+      refs.put(name, snapshotRef);
+      updatedBuilder.setBranch(name, snapshotRef);
```

Comment:
Should be updated to call setBranch depending on if it's a branch or setTag if it's a tag